dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipUdpConnection.java

```diff
@@ -0,0 +1,119 @@
+import com.ibm.ws.sip.stack.context.MessageContext;
+import com.ibm.ws.sip.stack.transaction.transport.UseCompactHeaders;
+import com.ibm.ws.sip.stack.transaction.transport.connections.SIPListenningConnection;
+//TODO Liberty import com.ibm.ws.management.AdminHelper;
+
+import io.netty.channel.Channel;
+
+/**
+ * a UDP connection object.
+ * there is no actual connection in UDP, but this object is used for
+ * sending and receiving multiple datagrams with the same UDP endpoint
+ * 
+ * @author ran
+ */
+public class SipUdpConnection extends BaseConnection
+{
+	/** the connection link for sending outbound messages */
+	private UdpSender m_connLink;
+	
+	/** a Netty channel associated with this SIP connection  */
+	private Channel m_channel;
+	
+	/**
+	 * constructor for inbound channels
+	 * @param lc channel that created this connection
+	 * @param connLink the connection link for sending outbound messages
+	 * @param channel a Netty channel associated with this SIP connection 
+	 */
+	public SipUdpConnection(SIPListenningConnection lc, UdpSender connLink, Channel channel) {
+		this(null, 0, lc, connLink, channel);
+	}
+	
+	/**
+	 * constructor
+	 * @param peerHost remote host address in dotted form, null if inbound
+	 * @param peerPort remote port number, 0 if inbound
+	 * @param channel channel that created this connection
+	 * @param connLink the connection link for sending outbound messages
+	 * @param channel a Netty channel associated with this SIP connection 
+	 */
+	public SipUdpConnection(String peerHost, int peerPort, SIPListenningConnection lc, UdpSender connLink, Channel channel) {
+		super(peerHost, peerPort, lc);
+		m_connLink = connLink;
+		m_channel = channel;
+	}
+	
+	/**
+	 * @see com.ibm.ws.sip.stack.transaction.transport.connections.channelframework.BaseConnection#write(com.ibm.ws.sip.stack.transaction.transport.connections.SipMessageByteBuffer, boolean, UseCompactHeaders)
+	 */
+	public void write(MessageContext messageSendingContext, boolean considerMtu, UseCompactHeaders useCompactHeaders) throws IOException {
+		prepareBuffer(messageSendingContext,considerMtu,useCompactHeaders);
+		
+		// send the message, or queue it if cannot send right now
+		m_connLink.send(messageSendingContext, useCompactHeaders);
+	}
+
+	public Channel getChannel() {
+		return m_channel;
+	}
+	
+	/**
+	 * @see com.ibm.ws.sip.stack.transaction.transport.connections.SIPConnection#connect()
+	 */
+	public void connect() throws IOException {
+	}
+	
+	/**
+	 * @see com.ibm.ws.sip.stack.transaction.transport.connections.SIPConnection#getTransport()
+	 */
+	public String getTransport() {
+		return "udp";
+	}
+
+	/**
+	 * @see com.ibm.ws.sip.stack.transaction.transport.connections.SIPConnection#isReliable()
+	 */
+	public boolean isReliable() {
+		return false;
+	}
+
+	/**
+	 * @see com.ibm.ws.sip.stack.transaction.transport.connections.SIPConnection#isSecure()
+	 */
+	public boolean isSecure() {
+		return false;
```

Comment:
There are lots of mentions of TLS above - is this finished?

Reply:
TLS can only run on top of TCP.  This, being a UDP connection, will never be secure.